app/controllers/tweets_controller.rb

```diff
@@ -0,0 +1,60 @@
+class TweetsController < ApplicationController
+  enable :sessions
+  set :session_secret, 'fwitter'
+  set :public_folder, 'public'
+  set :views, 'app/views'
+
+# Shows Tweets index page
+  get '/tweets' do
+    redirect '/login' if !logged_in?
+    @tweets = Tweet.all
+    erb :'/tweets/tweets'
+  end
+
+# Shows the "New Tweet" form and makes a new tweet
+  get '/tweets/new' do
+    redirect '/login' if !logged_in?
+    erb :'/tweets/new'
+  end
+
+  post '/tweets' do
+    redirect '/login' if !logged_in?
+    redirect '/tweets/new' if params[:tweet][:content].empty?
+    @tweet = Tweet.create(params[:tweet])
+    @tweet.user_id = User.find_by(username: current_user).id
+    @tweet.save
+    redirect "/tweets/#{@tweet.id}"
+  end
+
+# Shows an individual tweet
+  get '/tweets/:id' do
+    redirect '/login' if !logged_in?
+    @tweet = Tweet.find(params[:id])
+    erb :'/tweets/show'
+  end
+
+  #Deletes a tweet.
+  delete '/tweets/:id/delete' do
```

Comment:
Prefer to find tweets that are already only associated to the `current_user` than to manually check the `user_id` foreign key of the tweet against the `current_user.id`

```ruby
tweet = current_user.tweets.find_by(id: params[:id])
if tweet && tweet.destroy
  redirect "/tweets"
else
  redirect "/tweets/#{tweet.id}"
end
```

We're using the association collection method provided by [`has_many`](http://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#method-i-has_many) to query the user's tweets. 

Note the usage of [`find_by`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find-by) vs [`find`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find). Had we used `find` here, ActiveRecord would raise an error in the event that we could not find a tweet associated with this user. `find_by` will allow this action to fail gracefully, simply returning nil if the user is trying to delete a tweet that they do not have access to.